pkg/reconciler/queue_pipelineruns.go

```diff
@@ -34,6 +34,13 @@ func (r *Reconciler) queuePipelineRun(ctx context.Context, logger *zap.SugaredLo
 		return fmt.Errorf("updateError: %w", err)
 	}
 
+	// find global repository if set
+	globalRepo, err := r.repoLister.Repositories(r.run.Info.Kube.Namespace).Get(r.run.Info.Controller.GlobalRepository)
```

Comment:
do we need to fetch here, it is in reconciler right?

Reply:
yeah, it is reconciler.